python/cuml/cuml/cluster/hdbscan/hdbscan.pyx

```diff
@@ -831,6 +856,35 @@ class HDBSCAN(UniversalBase, ClusterMixin, CMajorInputTagMixin):
                 raise ValueError("Cluster selection method not supported. "
                                  "Must one of {'eom', 'leaf'}")
 
+            if self.build_algo == "auto":
+                if self.n_rows <= 50000:
+                    # brute force is faster for small datasets
+                    logger.warn("Building knn graph using brute force")
+                    self.build_algo = "brute_force_knn"
+                else:
+                    logger.warn("Building knn graph using nn descent")
+                    self.build_algo = "nn_descent"
+
+            if self.build_algo == 'brute_force_knn':
+                params.build_algo = GRAPH_BUILD_ALGO.BRUTE_FORCE_KNN
+            elif self.build_algo == 'nn_descent':
+                params.build_algo = GRAPH_BUILD_ALGO.NN_DESCENT
+                if self.build_kwds is None:
+                    params.nn_descent_params.graph_degree = <size_t> 64
+                    params.nn_descent_params.intermediate_graph_degree = <size_t> 128
+                    params.nn_descent_params.max_iterations = <size_t> 20
+                    params.nn_descent_params.termination_threshold = <float> 0.0001
+                    params.nn_descent_params.return_distances = <bool> True
+                else:
+                    params.nn_descent_params.graph_degree = <size_t> self.build_kwds.get("nnd_graph_degree", 64)
+                    params.nn_descent_params.intermediate_graph_degree = <size_t> self.build_kwds.get("nnd_intermediate_graph_degree", 128)
+                    params.nn_descent_params.max_iterations = <size_t> self.build_kwds.get("nnd_max_iterations", 20)
+                    params.nn_descent_params.termination_threshold = <float> self.build_kwds.get("nnd_termination_threshold", 0.0001)
+                    params.nn_descent_params.return_distances = <bool> self.build_kwds.get("nnd_return_distances", True)
+            else:
+                raise ValueError("Build algo not supported. "
+                                 "Must one of {'brute_force_knn', 'nn_descent'}")
+
```

Comment:
Would be nice to create a function to avoid repeating this section of code.

Reply:
100% agree w/ @viclafargue here.